eclipse-che/che-plugin-registry
# Enable subpath mode for Che Theia editor in devworkspaces

### What does this PR do?
adds urlRewriteSupported attribute for che theia endpoints so that they can be served on subpath instead of subdomain
also adds this attribute to che-machine-exec endpoint which is not mentioned in the referenced issue below

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20180

### How to test this PR?

1. create patch.yaml: 
```shell
echo \
"spec:
  server:
    pluginRegistryImage: quay.io/pbaran/che-plugin-registry:singlehost-theia
    pluginRegistryPullPolicy: Always" \
> patch.yaml
```
2. deploy che:
```shell
  chectl server:deploy --platform=openshift --workspace-engine=dev-workspace --che-operator-cr-patch-yaml=patch.yaml 
```
3. create workspace from quarkus sample
4. run:
```shell
kubectl get devworkspace --namespace=${USERNAMESPACE}
```
verify that devworkspace info shows theia serving on subpath

5. run:
```shell
kubectl describe configmaps --namespace=eclipse-che
```
verify that workspace configmap contains endpoints with subpath url

6. run:
```shell
kubectl get route --namespace=${USERNAMESPACE}
```
verify that there arent routes for che theia

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
